client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -118,23 +121,27 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		const planSlug = getUpgradePlanSlugFromPath( this.props.product, this.props.selectedSite );
-
-		let cartItem,
-			cartMeta;
-
-		if ( planSlug ) {
-			cartItem = getCartItemForPlan( planSlug );
+		const { product, productsList, purchaseId, selectedSite, selectedSiteSlug } = this.props;
+		const meta = product.split( ':' )[ 1 ];
+		const productSlug = getUpgradePlanSlugFromPath( product, selectedSite.jetpack ) || product.split( ':' )[ 0 ];
+
+		let cartItem = getCartItemForPlan( productSlug );
+
+		if ( 'theme' === productSlug ) {
+			cartItem = themeItem( meta );
+		} else if ( 'domain-mapping' === productSlug ) {
+			cartItem = domainMapping( { domain: meta } );
+		} else if ( 'no-ads' === productSlug ) {
```

Comment:
No Ads is a legacy product that is no longer available. We never converted any existing users to a plan, unlike what we did with other legacy products. Unlike the other products, it has an inconvenient slug that does not represent the product but something else.

I'll update the testing instructions.